core/templates/dev/head/pages/topic_editor/questions/QuestionsTabDirective.js

```diff
@@ -49,23 +53,76 @@ oppia.directive('questionsTab', [
             $scope.questionSuggestionThreads = [];
             $scope.activeQuestion = null;
             $scope.suggestionReviewMessage = null;
+            $scope.questionIsBeingUpdated = false;
+            $scope.questionIsBeingSaved = false;
           };
 
-          $scope.saveAndPublishQuestion = function() {
+          $scope.hasChanges = function() {
+            return QuestionUndoRedoService.hasChanges();
+          };
+
+          $scope.saveQuestion = function() {
             var validationErrors = $scope.question.validate(
               $scope.misconceptions);
             if (validationErrors) {
               AlertsService.addWarning(validationErrors);
               return;
             }
-            EditableQuestionBackendApiService.createQuestion(
-              $scope.skillId, $scope.question.toBackendDict(true)
-            ).then(function() {
-              TopicEditorStateService.fetchQuestionSummaries(
-                $scope.topic.getId(), function() {
-                  _initTab();
-                }
-              );
+            if (!$scope.questionIsBeingUpdated) {
+              $scope.questionIsBeingSaved = true;
+              EditableQuestionBackendApiService.createQuestion(
+                $scope.skillId,
+                $scope.question.toBackendDict(true)
+              ).then(function() {
+                TopicEditorStateService.fetchQuestionSummaries(
+                  $scope.topic.getId(), function() {
+                    _initTab();
+                  }
+                );
+                $scope.questionIsBeingSaved = false;
+              });
+            } else {
+              if (QuestionUndoRedoService.hasChanges()) {
+                $scope.questionIsBeingSaved = true;
+                // TODO(tjiang11): Allow user to specify a commit message.
```

Comment:
Is this tracked anywhere (in an issue perhaps)? Presumably need to ensure this is done prior to the feature being launched.

Reply:
Filed as #5690 (planning to do this next while submitting my PR for adding the questions tab to the skill editor)